src/components/PeopleFilters/PeopleFilters.tsx

```diff
@@ -0,0 +1,124 @@
+import { ChangeEvent, FC, memo } from 'react';
+import { useSearchParams } from 'react-router-dom';
+import cn from 'classnames';
+
+import { getSearchWith } from '../../utils/searchHelper';
+import { SearchLink } from '../SearchLink/SearchLink';
+
+const sexItems = [
+  { sex: '', label: 'All' },
+  { sex: 'm', label: 'Male' },
+  { sex: 'f', label: 'Female' },
+];
+
+const centuriesItems = ['16', '17', '18', '19', '20'];
+
+export const PeopleFilters: FC = memo(() => {
+  const [searchParams, setSearchParams] = useSearchParams();
+  const query = searchParams.get('query') || '';
+  const sex = searchParams.get('sex') || '';
+  const centuries = searchParams.getAll('centuries') || [];
+
+  const onQueryChange = (event: ChangeEvent<HTMLInputElement>) => {
+    const updatedSearch = getSearchWith(
+      searchParams, { query: event.target.value || null },
+    );
+
+    setSearchParams(updatedSearch);
+  };
+
+  const changeSexParams = (item: string) => ({
+    sex: item || null,
+  });
+
+  const changeCenturyParams = (century: string) => ({
+    centuries: centuries.includes(century)
+      ? centuries.filter(c => c !== century)
+      : [...centuries, century],
+  });
+
+  const resetCenturyParams = () => ({
+    centuries: null,
+  });
+
+  const resetAllParams = () => ({
```

Comment:
Do not change the parameters manually, use the `state` component for this.
This applies to all parameters and all changes to the component.